dask/array/routines.py

```diff
@@ -515,34 +515,42 @@ def gradient(f, *varargs, **kwargs):
     return results
 
 
+def _bincount_sum(bincounts, dtype=int):
+    n = max(map(len, bincounts))
+    out = np.zeros(n, dtype=dtype)
+    for b in bincounts:
+        out[:len(b)] += b[:len(b)]
+    return out
+
+
 @wraps(np.bincount)
-def bincount(x, weights=None, minlength=None):
-    if minlength is None:
-        raise TypeError("Must specify minlength argument in da.bincount")
+def bincount(x, weights=None, minlength=0):
     assert x.ndim == 1
     if weights is not None:
         assert weights.chunks == x.chunks
 
     # Call np.bincount on each block, possibly with weights
+    dsk = {}
     token = tokenize(x, weights, minlength)
-    name = 'bincount-' + token
-    if weights is not None:
-        dsk = {(name, i): (np.bincount, (x.name, i), (weights.name, i), minlength)
-               for i, _ in enumerate(x.__dask_keys__())}
-        dtype = np.bincount([1], weights=[1]).dtype
-    else:
-        dsk = {(name, i): (np.bincount, (x.name, i), None, minlength)
-               for i, _ in enumerate(x.__dask_keys__())}
-        dtype = np.bincount([]).dtype
-
-    # Sum up all of the intermediate bincounts per block
     name = 'bincount-sum-' + token
-    dsk[(name, 0)] = (np.sum, list(dsk), 0)
-
-    chunks = ((minlength,),)
+    for i, _ in enumerate(x.__dask_keys__()):
+        if weights is not None:
+            dsk[('bincount-' + token, i)] = (np.bincount, (x.name, i),
+                                             (weights.name, i), minlength)
+            dtype = np.bincount([1], weights=[1]).dtype
+        else:
+            dsk[('bincount-' + token, i)] = (np.bincount, (x.name, i),
+                                             None, minlength)
+            dtype = np.bincount([]).dtype
```

Comment:
What's the reason to move the `for` outside of the `if`?

Reply:
No real reason, just thought it was clearer to read at one stage. These could probably go back to being list comprehensions now.